python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1143,6 +1190,120 @@ def _replace_closurevars(
     )
 
 
+_TYPES: Set[str] = {
```

Comment:
*suggestion*: I don't think these should live in the generic fast_slow_proxy code.

maintaining this list is also error-prone: is there some way we can populate it automagically?

My suggestion would be that the `assert_func` encodes the skipping logic. So for the pandas-specific stuff we would do something like:

```
def assert_func(got, expect):
    if should_check(got):
        assert_eq(got, expect)
    return None
```

And `should_check` can encode the specific information.

Reply:
I agree that we don't want to maintain an explicit list of types here. I _think_ you want to use `_is_final_type` to determine whether to execute the check. I don't think it makes sense to validate intermediate types.

https://github.com/rapidsai/cudf/blob/ff981a4048a389b0e2582e94d3397a83096d16c9/python/cudf/cudf/pandas/fast_slow_proxy.py#L1060-L1061

Reply:
@bdice `get_final_type_map` is filled with type names after `_fast_slow_function_call` is called in [`make_final_proxy_type`](https://github.com/rapidsai/cudf/pull/15837/files#diff-2b6c5fdc63e639f8422924db13aed5ddc034848825aac4b4077df7d49c79cfaeR184). How should I handle this?

Reply:
That call isn't actually happening in `make_final_proxy_type`. `make_final_proxy_type` is defining an `__init__` function that calls `_fast_slow_function_call`, then that `__init__` function is set as the `__init__` method of the new type that's being created. This means that at runtime, construction of a wrapped type will follow the fast slow path for constructing the underlying type. For example, doing `df = pd.DataFrame(*args, **kwargs)` will run the body of this `__init__` function and trigger a `_fast_slow_function_call` that will first invoke the cudf DataFrame constructor, then if that fails the real pandas DataFrame constructor.

Does that address your concern?